Framework/Algorithms/src/RadiusSum.cpp

```diff
@@ -619,16 +613,16 @@ void RadiusSum::setUpOutputWorkspace(std::vector<double> &values) {
       inputWS, 1, values.size() + 1, values.size());
 
   g_log.debug() << "Set the data\n";
-  MantidVec &refY = outputWS->dataY(0);
-  std::copy(values.begin(), values.end(), refY.begin());
+  outputWS->mutableY(0) = values;
 
   g_log.debug() << "Set the bins limits\n";
-  MantidVec &refX = outputWS->dataX(0);
+
+  auto xSize = outputWS->mutableX(0).size();
   double bin_size = (max_radius - min_radius) / num_bins;
+  outputWS->setBinEdges(0, xSize,
+                        HistogramData::LinearGenerator(min_radius, bin_size));
 
-  for (int i = 0; i < (static_cast<int>(refX.size())) - 1; i++)
-    refX[i] = min_radius + i * bin_size;
-  refX.back() = max_radius;
+  outputWS->mutableX(0).back() = max_radius;
```

Comment:
@SimonHeybrock Do you think this is not needed anymore because of `LinearGenerator`, I ran the tests with both and it passes with both, but decided to leave it just in case, as it seems that was the intention before?